soc/arm/nxp_imx/rt/soc_rt10xx.c

```diff
@@ -326,6 +306,19 @@ static int imxrt_init(const struct device *arg)
 	return 0;
 }
 
+#ifdef CONFIG_SOC_MIMXRT1011
+void SystemInitHook(void)
+{
+	/* When set this bit, FlexSPI will fetch more data than AHB burst
+	 * required to meet the alignment requirement.
+	 * Without this, the FlexSPI will return corrupted data during early
+	 * boot, causing a hardfault. This can also be resolved by enabling
+	 * the instruction cache in very early boot.
+	 */
+	FLEXSPI->AHBCR |= FLEXSPI_AHBCR_READADDROPT_MASK;
```

Comment:
Setting this bit in the FlexSPI is specific to the external memory used, and should not depend on CONFIG_SOC_MIMXRT1011.  If the RT1011 device is used on a different board with different memory, that board may require READADDROPT to be cleared.  I think setting this bit should be in the mimxrt1010_evk board files.  I would also add in the comments that this bit is required for the memory device AT25SF128A, in case a custom board uses different memory.

Also, setting this bit early in startup code like this should only be done when the device boots and XIPs from FlexSPI flash.  If the ROM boots from a different boot option and does not initialize the clocks to the FlexSPI peripheral, then writing this register could lead to a bus fault.  The MCUXpresso SDK only sets this bit when `XIP_EXTERNAL_FLASH == 1`.  I think we have similar Kconfig symbols we could use here.